mindmeld/components/dialogue.py

```diff
@@ -419,7 +522,10 @@ def _set_target_state(request, responder):
 
         self._entrance_handler = _async_set_target_state if self.async_mode else _set_target_state
         app.add_dialogue_rule(self.name, self._entrance_handler, **kwargs)
-        handler = self.apply_handler_async if self.async_mode else self.apply_handler
+        if self.async_mode:
+            handler = self._apply_flow_handler_async
+        else:
+            handler = self._apply_flow_handler_sync
```

Comment:
Use the code pattern similar to before: `handler = self. _apply_flow_handler_async if self.async_mode else self. _apply_flow_handler_sync `